File: backend/core/src/main/java/site/timecapsulearchive/core/domain/capsule/data/secret_c/SecretCapsuleCreateRequestDto.java
package site.timecapsulearchive.core.domain.capsule.data.secret_c;

import java.time.ZonedDateTime;
import java.util.List;
import lombok.Builder;
import site.timecapsulearchive.core.domain.capsule.data.dto.AddressData;

@Builder
public record SecretCapsuleCreateRequestDto(
    Long capsuleSkinId,
    String title,
    String content,
    double longitude,
    double latitude,
    AddressData addressData,
    ZonedDateTime dueDate,
    List<String> imageNames,
    List<String> videoNames,
    String directory
) {

}

File: frontend/ARchive/app/src/main/java/com/droidblossom/archive/data/dto/secret/request/SecretCapsuleCreateRequestDto.kt
package com.droidblossom.archive.data.dto.secret.request

import com.droidblossom.archive.data.dto.capsule.response.AddressDataDto
import com.droidblossom.archive.data.dto.common.FileNameDto

data class SecretCapsuleCreateRequestDto(
    val capsuleSkinId: Long,
    val content: String,
    val directory: String,
    val dueDate: String?,
    val imageNames: List<String>?,
    val videoNames: List<String>?,
    val addressData: AddressDataDto,
    val latitude: Double,
    val longitude: Double,
    val title: String
)